internal/services/data_record/resource_data_record_test.go

```diff
@@ -1361,3 +1363,227 @@ func TestUnitDataRecordResource_Validate_Update_Relationships(t *testing.T) {
+		Steps: []resource.TestStep{
+			{
+				Config: `
+				resource "powerplatform_data_record" "contant" {
+					disable_on_destroy = true
+					environment_id     = "00000000-0000-0000-0000-000000000001"
+					table_logical_name = "contact"
+					columns = {
+					  firstname          = "John"
+					  lastname           = "Doe"
+					}
+				}
+				
+				resource "powerplatform_data_record" "business_unit" {
+					environment_id     = "00000000-0000-0000-0000-000000000001"
+					table_logical_name = "businessunit"
+					disable_on_destroy = true
+					columns = {
+						name       = "test"
+					}
+				}`,
+			},
+			{
+				// commenting out the resource to trigger the delete
+				Config: ` 
+				// resource "powerplatform_data_record" "contant" {
+				// 	disable_on_destroy = true
+				// 	environment_id     = "00000000-0000-0000-0000-000000000001"
+				// 	table_logical_name = "contact"
+				// 	columns = {
+				// 	  firstname          = "John"
+				// 	  lastname           = "Doe"
+				// 	}
+				// }
+				
+				// resource "powerplatform_data_record" "business_unit" {
+				// 	environment_id     = "00000000-0000-0000-0000-000000000001"
+				// 	table_logical_name = "businessunit"
+				// 	disable_on_destroy = true
+				// 	columns = {
+				// 		name       = "test"
+				// 	}
+				// }`,
+				Check: resource.ComposeTestCheckFunc(
+					func(_ *terraform.State) error {
+						if !contactDisabledExecuted {
+							return fmt.Errorf("expected contact to be disabled on delete")
+						}
+						return nil
+					},
+					func(_ *terraform.State) error {
+						if !businessUnitDisabledExecuted {
+							return fmt.Errorf("expected business unit to be disabled on delete")
+						}
+						return nil
+					},
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataRecordResource_Validate_Disable_On_Delete(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		IsUnitTest:               true,
+		ProtoV6ProviderFactories: mocks.TestUnitTestProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: `
+				resource "powerplatform_environment" "test_env" {
+					display_name     = "` + mocks.TestName() + `"
+					location         = "unitedstates"
+					environment_type = "Sandbox"
+					dataverse = {
+					  language_code     = "1033"
+					  currency_code     = "USD"
+					  security_group_id = "00000000-0000-0000-0000-000000000000"
+					}
+				}
+
+				data "powerplatform_data_records" "root_business_unit" {
+					environment_id    = powerplatform_environment.test_env.id
+					entity_collection = "businessunits"
+					filter            = "parentbusinessunitid eq null"
+					select            = ["name"]
+				}
+
+				resource "powerplatform_data_record" "contant" {
+					disable_on_destroy = true
+					environment_id     = powerplatform_environment.test_env.id
+					table_logical_name = "contact"
+					columns = {
+					  firstname          = "John"
+					  lastname           = "Doe"
+					}
+				}
```

Comment:
pick a table that is not contact.  we explicitly say not to use this resource with contact.  We should test with other config tables.

Reply:
app user would be a good one to add to the test

Reply:
done